cli/rt/41_prompt.js

```diff
@@ -0,0 +1,66 @@
+// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+((window) => {
+  const { stdin, stdout } = window.__bootstrap.files;
+  const { isatty } = window.__bootstrap.tty;
+  const LF = "\n".charCodeAt(0);
+  const encoder = new TextEncoder();
+  const decoder = new TextDecoder();
+
+  function alert(message = "Alert") {
+    if (!isatty(stdin.rid)) {
+      return;
+    }
+
+    stdout.writeSync(encoder.encode(`${message} [Enter] `));
+
+    readLineFromStdinSync();
+  }
+
+  function confirm(message = "Confirm") {
+    if (!isatty(stdin.rid)) {
+      return false;
+    }
+
+    stdout.writeSync(encoder.encode(`${message} [y/N] `));
+
+    const answer = readLineFromStdinSync();
+
+    return answer === "Y" || answer === "y";
+  }
+
+  function prompt(message = "Prompt", defaultValue) {
+    defaultValue ??= null;
+
+    if (!isatty(stdin.rid)) {
+      return null;
+    }
+
+    stdout.writeSync(encoder.encode(`${message} `));
+
+    if (defaultValue) {
+      stdout.writeSync(encoder.encode(`[${defaultValue}] `));
+    }
+
+    return readLineFromStdinSync() || defaultValue;
+  }
+
+  function readLineFromStdinSync() {
+    const c = new Uint8Array(1);
+    const buf = [];
+
+    while (true) {
+      const n = stdin.readSync(c);
+      if (n === 0 || c[0] === LF) {
+        break;
+      }
+      buf.push(c[0]);
+    }
+    return decoder.decode(new Uint8Array(buf));
+  }
```

Comment:
How will this function behave if I type `yes` before pressing enter?

Reply:
Currently only `y` and `Y` are interpreted as `true`. But maybe we should consider words like `yes`, `Yes`, `YES` as true as well?

Reply:
I'm more concerned that typing `yes` and hitting enter will cause the first prompt to get "true" value, and subsequent prompts would read `e` and `s` - I think we should read the whole line instead of a single char.

Reply:
The above while-loop reads from stdin until it finds LF. So it reads the entire single line. The pattern check against `y` or `Y` is done in the above `confirm` function (it checks against the entire line).

Reply:
Great!